tests/finance/test_commissions.py

```diff
@@ -79,6 +79,16 @@ def verify_per_trade_commissions(self,
         self.assertEqual(0, model.calculate(order, txns[1]))
         self.assertEqual(0, model.calculate(order, txns[2]))
 
+    def test_equality_and_comparison(self):
+        perdollar1 = PerDollar(cost=0.2)
+        perdollar2 = PerDollar(cost=0.2)
+
+        self.assertEqual(perdollar1, perdollar2)
+        self.assertEqual(hash(perdollar1), hash(perdollar2))
+
+        self.assertEqual(perdollar1.__dict__, perdollar1.asdict())
+        self.assertEqual(perdollar2.__dict__, perdollar2.asdict())
```

Comment:
My instinct is to remove these last two assertions, since they imply implementation requirements - as opposed to interface requirements, which you're checking above. I could see asserting that the return value is `{'cost_per_dollar': ...}`, but I'd probably remove it instead, since `asdict` is really more of an implementation detail of `__eq__` and `__hash__`.